File: fast_python_numba/is_prime.py
import math
import timeit

def is_prime(number):
    if number == 2:
        return True
    if number <= 1 or not number % 2:
        return False
    max_range = int(math.sqrt(number)) + 1
    for div in range(3, max_range, 2):
        if not number % div:
            return False
    return True


def run_program(max_number):
    for number in range(max_number):
        is_prime(number)


if __name__ == '__main__':
    max_number = 10000000
    start = timeit.default_timer()
    run_program(max_number)
    stop = timeit.default_timer()
    print(stop - start, " (seconds)")

File: fast_python_numba/is_prime.cpp
#include <iostream>
#include <cmath>
#include <time.h>

using namespace std;

bool isPrime(int number)
{
    if (number == 2)
        return true;
    if (number <= 1 || number % 2 == 0)
        return false;
    double sqrt_num = sqrt(double(number));
    for (int div = 3; div <= sqrt_num; div += 2)
    {
        if (number % div == 0)
            return false;
    }
    return true;
}

void runProgram(int max_number)
{
    for (int number = 0; number < max_number; number++)
        isPrime(number);
}

int main()
{
    int MAX_NUMBER = 10000000;
    clock_t start, end;
    start = clock();
    runProgram(MAX_NUMBER);
    end = clock();
    cout << (end - start) / ((double)CLOCKS_PER_SEC);
    cout << " (seconds)\n";
    return 0;
}